components/playlist/browser/resources/reducers/playlist_reducer.ts

```diff
@@ -50,9 +51,21 @@ const playlistReducer: Reducer<PlaylistData | undefined> = (
       state = { ...state, currentList, lists: [...playlists] }
       break
 
-    case types.SELECTED_PLAYLIST_UPDATED:
-      const playlist = action.payload
-      state = { ...state, currentList: playlist }
+    case types.PLAYLIST_UPDATED:
+      const updatedPlaylist = action.payload
+      state = { ...state }
+      if (state.currentList?.id === updatedPlaylist.id) {
+        state.currentList = updatedPlaylist
+      }
+
+      const oldPlaylist = state.lists.find(e => {
```

Comment:
Nit: You can use `findIndex`  directly, instead of `find` and then `indexOf`

Reply:
Oh cool! Thanks.